bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,377 @@
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, Union
+
+from async_rediscache import RedisCache
+from discord import Embed, Message, Reaction, TextChannel, User
+from discord.ext import tasks
+from discord.ext.commands import Bot, Cog, Context, group
+
+from bot.constants import Channels, Client, Colours, Month
+from bot.utils.decorators import InMonthCheckFailure
+
+logger = logging.getLogger(__name__)
+MONTH_TO_LIMIT_TO = Month.OCTOBER
+
+EMOJIS_VAL = {  # the value of the emojis
+    '\N{Jack-O-Lantern}': 1,
+    '\N{Ghost}': 2,
+    '\N{Skull and Crossbones}': 3,
+    '\N{Zombie}': 4,
+    '\N{Face Screaming In Fear}': 5
+}
+
+
+class SpookNameRate(Cog):
+    """
+    A game that asks the user to spookify or halloweenify a name that is given everyday.
+
+    It sends a random name everyday. The user needs to try and spookify it to his best ability and
+    send that name back using the `spooknamerate add entry` command
+    """
+
+    messages = RedisCache()  # a variable to store all the messages
+    data = RedisCache()
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+        names_data = self.load_json(Path('bot', 'resources', 'halloween', 'spooknamerate_names.json'))
+        self.first_names = names_data['first_names']
+        self.last_names = names_data['last_names']
+        # the names are from https://www.mockaroo.com/
+
+        self.name = None
+
+        self.added_messages = ["Let's see if you win?", ":jack_o_lantern: SPOOKY :jack_o_lantern:",
+                               "If you got it, haunt it.", "TIME TO GET YOUR SPOOK ON! :skull:"]
+
+        self.bot.loop.create_task(self.load_vars())
+
+        self.first_time = None
+        self.poll = False
+        self.channel_id = Channels.community_bot_commands
+        self.announce_name.start()
+        self.checking_messages = asyncio.Lock()  # Define an asyncio.Lock() to make sure the dictionary isn't changed
+        # when checking the messages for duplicate emojis'
+
+    async def load_vars(self) -> None:
+        """Loads the variables that couldn't be loaded in __init__."""
+        self.first_time = await self.data.get('first_time', True)
+        self.name = await self.data.get('name')
+
+    @group(name="spooknamerate", invoke_without_command=True)
+    async def spook_name_rate(self, ctx: Context) -> None:
+        """Get help on the Spook Name Rate game."""
+        help_embed = Embed(
+            title="Spook Name Rate",
+            description=f"Help for the `{self.bot.command_prefix}spooknamerate` command",
+            color=Colours.soft_orange,
+        )
+
+        emoji_message = "\n".join([f"- {emoji} {val}" for emoji, val in EMOJIS_VAL.items()])
+
+        help_embed.add_field(
+            name="How to play",  # display messages on HOW TO PLAY
+            value=f"""Everyday, the bot will post a random name, which you will need to spookify using your creativity.
```

Comment:
Let's move this message to constant.

Reply:
The entire `Embed`, right?

Reply:
Actually, this should use JSON for this help message. Take a look to `AdventOfCode` cog that uses JSON to store help messages.

Reply:
I don't think I can do that. Currently I am dynamically producing some data using a variable in my program (`EMOJIS_VAL`) which I don't think I should hardcode.

Reply:
Okay, then make content (not field titles) constants.

Reply:
Okay sure!

Reply:
I have moved everything to be constant, is it better now @ks129 ?

Reply:
Much better.